californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -358,6 +381,52 @@ public synchronized void stop() {
 		}
 	}
 
+	/**
+	 * Dump exchanges of stores.
+	 * 
+	 * @param logMaxExchanges maximum number of exchanges to include in dump.
+	 */
+	public void dump(int logMaxExchanges) {
+		if (LOGGER.isInfoEnabled()) {
+			LOGGER.info(dumpCurrentLoadLevels());
+			if (0 < logMaxExchanges) {
+				if (!exchangesByMID.isEmpty()) {
+					dumpExchanges(logMaxExchanges, exchangesByMID.entrySet());
+				}
+				if (!exchangesByToken.isEmpty()) {
+					dumpExchanges(logMaxExchanges, exchangesByToken.entrySet());
+				}
+			}
+		}
+	}
+
+	/**
+	 * Dump collection of exchange entries.
+	 * 
+	 * @param logMaxExchanges maximum number of exchanges to include in dump.
+	 * @param exchangeEntries collection with exchanges entries
+	 */
+	private <K> void dumpExchanges(int logMaxExchanges, Set<Entry<K, Exchange>> exchangeEntries) {
+		for (Entry<K, Exchange> exchangeEntry : exchangeEntries) {
```

Comment:
shouldn't this be guarded by a check for the log level as well?

Reply:
The method is not intended to be called outside of the dump().